src/devices/Shtc3/Shtc3.cs

```diff
@@ -0,0 +1,290 @@
+        public bool TryGetTempAndHumi(out Measure sensorMeasure, bool LowPower = false, bool ClockStretching = false)
+        {
+            Register cmd = GetMeasurementCmd(LowPower, ClockStretching);
+
+            if (!TryReadSensorData(cmd, out var st, out var srh))
+            {
+                sensorMeasure = default(Measure);
+                return false;
+            }
+
+            // Details in the Datasheet P9
+            sensorMeasure = new Measure(Temperature.FromCelsius(Math.Round(st * 175 / 65536.0 - 45, 1)),
+                                                 Math.Round(srh * 100 / 65536.0, 1));
+            return true;
+        }
+
+        private bool TryReadSensorData(Register cmd, out double temperature, out double humidity)
+        {
+            Write(cmd);
+
+            Span<byte> readBuff = stackalloc byte[6];
+
+            _i2cDevice.Read(readBuff);
+
+            // Details in the Datasheet P7
+            int st = (readBuff[0] << 8) | readBuff[1];      // Temp
+            int srh = (readBuff[3] << 8) | readBuff[4];     // Humi
+
+            // check 8-bit crc
+            bool tCrc = CheckCrc8(readBuff.Slice(0, 2), readBuff[2]);
+            bool rhCrc = CheckCrc8(readBuff.Slice(3, 2), readBuff[5]);
+            if (tCrc == false || rhCrc == false)
+            {
+                temperature = double.NaN;
+                humidity = double.NaN;
+                return false;
+            }
+
+            temperature = st;
+            humidity = srh;
+            return true;
+        }
+
+        private static Register GetMeasurementCmd(bool LowPower, bool ClockStretching)
+        {
+            Register mea = Register.SHTC3_MEAS_T_RH_POLLING_NPM;
+            if (LowPower)
+            {
+                if (ClockStretching)
+                {
+                    mea = Register.SHTC3_MEAS_T_RH_CLOCKSTR_LPM;
+                }
+                else
+                {
+                    mea = Register.SHTC3_MEAS_T_RH_POLLING_LPM;
+                }
+            }
+            else
+            {
+                if (ClockStretching)
+                {
+                    mea = Register.SHTC3_MEAS_T_RH_CLOCKSTR_NPM;
+                }
+            }
+
+            return mea;
+        }
+
+        /// <summary>
+        /// SHTC3 Sleep
+        /// </summary>
+        private void Sleep()
+        {
+            Write(Register.SHTC3_SLEEP);
+        }
+
+        /// <summary>
+        /// SHTC3 Wakeup
+        /// </summary>
+        private void Wakeup()
+        {
+            Write(Register.SHTC3_WAKEUP);
+        }
+
+        /// <summary>
+        /// SHTC3 Soft Reset
+        /// </summary>
+        public void Reset()
+        {
+            Write(Register.SHTC3_RESET);
+        }
+
+        private int _id = 0;
+
+        /// <summary>
+        /// Sensor Id
+        /// </summary>
+        public int Id
+        {
+            get
+            {
+                if (_id == 0)
```

Comment:
is 0 invalid number? if not then use different value (perhaps -1, if all values are fine as ID then use `int?`)